bot/cogs/tags.py

```diff
@@ -27,6 +26,44 @@ def __init__(self, bot: Bot):
         self.bot = bot
         self.tag_cooldowns = {}
 
+        self._cache = {}
+        self._last_fetch = None
+
+    async def _get_tags(self, is_forced: bool = False) -> None:
+        """Getting all tags."""
+        # Refresh only when there's a more than 5m gap from last call.
+        if is_forced or not self._last_fetch or time.time() - self._last_fetch > 5 * 60:
+            tags = await self.bot.api_client.get('bot/tags')
+            self._cache = {tag['title'].lower(): tag for tag in tags}
+
+    @staticmethod
+    def _fuzzy_search(search: str, target: str) -> bool:
+        found = 0
+        index = 0
+        _search = search.lower().replace(' ', '')
+        _target = target.lower().replace(' ', '')
+        for letter in _search:
+            index = _target.find(letter, index)
+            if index == -1:
+                break
+            found += index > 0
+        return found / len(_search) * 100
+
+    def _get_suggestions(self, tag_name: str, score: int) -> list:
+        return sorted(
+            (tag for tag in self._cache.values() if Tags._fuzzy_search(tag_name, tag['title']) >= score),
+            key=lambda tag: Tags._fuzzy_search(tag_name, tag['title']),
+            reverse=True
+        )
```

Comment:
Calling `_fuzzy_search` multiple times per tag is pretty inefficient. The scores could be calculated ones and stored in a dictionary e.g. `dict[tag] = score` which could be used in the key function.

Reply:
That is a great suggestions! I have created a dictionary to store the scores in the form of `tag_title: matching score` ( str, int ) and by doing so, won't be calling `Tags._fuzzy_search()` in other places.